openapi/iris.helpdesk/client.bal

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/url;
+
+# Provides API key configurations needed when communicating with a remote HTTP endpoint.
+public type ApiKeysConfig record {|
+    # API keys related to connector authentication
+    map<string> apiKeys;
+|};
+
+# This is a generated connector for [IRIS Helpdesk API v1.6.4](https://www.iriscrm.com/api) OpenAPI specification.
+# Connect with the Helpdesk API to work with your ticket data.  
+# Helpdesk tickets are organized by ticket types.
+# Ticket types have properties such as assigned users, expected time to completion (SLA), and visibility to help you segment tickets based on job descriptions.
+# Get detailed helpdesk ticket information, perform updates, and manage attached files with ease by using the Helpdesk API.
+@display {label: "IRIS Helpdesk", iconPath: "resources/iris.helpdesk.svg"}
+public isolated client class Client {
+    final http:Client clientEp;
+    final readonly & map<string> apiKeys;
+    # Gets invoked to initialize the `connector`.
+    # The connector initialization requires setting the API credentials. 
+    # Create a [IRIS account](https://www.iriscrm.com) and obtain tokens by following [this guide](https://www.iriscrm.com/api/#section/Generate-an-API-token).
+    #
+    # + apiKeyConfig - Provide your API key as `X-API-KEY`. Eg: `{"X-API-KEY" : "<API key>"}` 
+    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ApiKeysConfig apiKeyConfig, http:ClientConfiguration clientConfig =  {}, string serviceUrl = "/api/v1") returns error? {
```

Comment:
The service URL given here `string serviceUrl = "/api/v1"` is an invalid URL since it doesn't have a valid domain and will fail at invocation. Please check.

Reply:
This is resolved now. Thank you.